src/index.js

```diff
@@ -485,7 +477,11 @@ class Libp2p extends EventEmitter {
         let discoveryService
 
         if (typeof DiscoveryService === 'function') {
-          discoveryService = new DiscoveryService(Object.assign({}, config, { peerInfo: this.peerInfo, libp2p: this }))
+          discoveryService = new DiscoveryService(Object.assign({}, config, {
+            peerId: this.peerId,
+            multiaddrs: this.addresses.listen,
```

Comment:
When we have an `AddressManager` the discovery service should just use that from `libp2p.addressManager`. While they don't right now, in the future our announce addresses will be able to change over time (AutoNat, AutoRelay, etc) and we should be sharing the most recent ones via the discovery services.

Reply:
Will include in #612  to remove this